batch7crud-roll3/src/main/java/com/lftechnology/batch7crud/dao/StudentDAO.java

```diff
@@ -0,0 +1,125 @@
+package com.lftechnology.batch7crud.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.lftechnology.batch7crud.constant.Query;
+import com.lftechnology.batch7crud.db.DBConnection;
+import com.lftechnology.batch7crud.entity.Student;
+import com.lftechnology.batch7crud.exception.DataException;
+
+public class StudentDAO {
+  private static final Logger LOGGER = Logger.getLogger("StudentDAO");
+
+  public void insert(Student student) throws DataException {
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement stmnt = conn.prepareStatement(Query.STUDENT_INSERT)) {
+      stmnt.setInt(1, student.getRoll());
+      stmnt.setString(2, student.getName());
+      stmnt.execute();
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public List<Student> fetch(int page, int pageSize) throws DataException {
+    List<Student> studentList = new ArrayList<Student>();
+
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement stmnt = conn.prepareStatement(Query.STUDENT_LIST);) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "" is provided externally to the method and not sanitized before use. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)